packages/gateway/src/manager.ts

```diff
@@ -271,7 +271,7 @@ export function createGatewayManager(options: CreateGatewayManagerOptions): Gate
       )
     },
     calculateWorkerId(shardId) {
-      const workerId = Math.min(shardId % gateway.shardsPerWorker, gateway.totalWorkers - 1)
+      const workerId = shardId % gateway.totalWorkers
```

Comment:
```suggestion
      const workerId = Math.min(Math.floor(shardId / manager.shardsPerWorker), manager.totalWorkers - 1);
```

Reply:
Is this the fixed formula? If so, what do you think if you just commit the suggestion and merge?